HW0/P4/P4.py

```diff
@@ -23,54 +23,148 @@
 
     # Create 3D axes for plotting
     ax = Axes3D(plt.figure())
-
     #####################
     # Part 1:
     #
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
+    s_true = np.loadtxt("P4_trajectory.txt", delimiter = ",")
+   
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    x_coords = s_true[:,0]
+    y_coords = s_true[:,1]
+    z_coords = s_true[:,2]
+   
+    ax.plot(x_coords, y_coords, z_coords,
+            '--b', label='True trajectory')
+    
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
-    #####################
+    m = np.loadtxt("P4_measurements.txt", delimiter = ",")
+    ###### print type(m)    
+    m_trans = np.transpose(m)
+
+    Cvtr = np.array([ [1/rx,0,0], [0,1/ry,0], [0,0,1/rz] ])
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    ppts = np.dot(Cvtr, m_trans)
+    ppts_trans = np.transpose(ppts)
+    #####################
+     
+    x_coords = ppts_trans[:,0]
+    y_coords = ppts_trans[:,1]
+    z_coords = ppts_trans[:,2]
+    ax.plot(x_coords, y_coords, z_coords,
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
-
+    A = np.array([ [1,0,0,dt,0,0],
+                   [0,1,0,0,dt,0],
+                   [0,0,1,0,0,dt],
+                   [0,0,0,1-c*dt,0,0],
+                   [0,0,0,0,1-c*dt,0],
+                   [0,0,0,0,0,1-c*dt] ])
+    a = np.array([ [0,0,0,0,0,g*dt] ]).T
+    s0 = np.array([ [0,0,2,15,3.5,4.0] ]).T
+    
     # Initial conditions for s0
     # Compute the rest of sk using Eq (1)
+    sk = s0
+    s  = s0
+
+    i = 1
+    while i < K:
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '-k', label='Blind trajectory')
+       sk = np.dot(A, sk) + a
+       s  = np.c_[s, sk]
+  
+       i += 1
+
+    ###### print s.shape  
+    x_coords = s[0,:] 
+    y_coords = s[1,:]
+    z_coords = s[2,:]
+    ax.plot(x_coords, y_coords, z_coords,
+            '-k', label='Blind trajectory')
 
     #####################
     # Part 4:
     # Use the Kalman filter for prediction
     #####################
 
-    # B = ?
-    # C = ?
+    B = np.array([ [bx,0,0,0,0,0],
+                   [0,by,0,0,0,0],
+                   [0,0,bz,0,0,0],
+                   [0,0,0,bvx,0,0],
+                   [0,0,0,0,bvy,0],
+                   [0,0,0,0,0,bvz] ])
+    
+    C = np.array([ [rx,0,0,0,0,0],
+                   [0,ry,0,0,0,0],
+                   [0,0,rz,0,0,0] ])
+    
```

Comment:
Again, I did these exactly the same way that you did, but I got these suggestions:
B = np.matrix(np.zeros([6,6]))
np.fill_diagonal(B,[bx,by,bz,bvx,bvy,bvz])

C = np.matrix(np.zeros([3,6]))
np.fill_diagonal(C,[rx,ry,rz])